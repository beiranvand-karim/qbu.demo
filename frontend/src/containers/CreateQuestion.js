import React, {Component} from 'react'

class CreateQuestion extends Component {
  render() {
    return (
      <>
        <h1>create question page</h1>
      </>
    )
  }
}

export default CreateQuestion;
