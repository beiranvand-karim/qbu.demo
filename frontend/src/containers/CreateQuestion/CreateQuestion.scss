
.createQuestionFormContainer {
  display: flex;
  justify-content: center;
  align-items: center;
}

.formGroup {
  &_input {
    width: 300px !important;
  }
}
