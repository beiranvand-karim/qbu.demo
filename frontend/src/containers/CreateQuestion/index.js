import React, {Component} from 'react'
import {Button, Form, FormGroup, Input, Label} from 'reactstrap'
import './CreateQuestion.scss'
import config from "../../config"
import {connect} from "react-redux"
import {
   createQuestionBegin,
   createQuestionError,
   createQuestionSuccess
} from '../../actions/CreateQuestionActions'

export class CreateQuestion extends Component {

   constructor() {
      super();
      this.createQuestion = this.createQuestion.bind(this);
      this.createQuestionFormSubmission = this.createQuestionFormSubmission.bind(this);
      this.change = this.change.bind(this)
   }

   createQuestion(question) {
      this.props.createQuestionBegin();
      fetch(`${config.server}/question`, {
         method: "POST",
         mode: "no-cors",
         headers: {
            "cache-control": "no-cache",
            "Content-Type": "application/json",
            "Authorization": "Basic " + btoa("demo" + ":" + "test2")
         },
         body: JSON.stringify(question),
      })
         .then(res => res.json())
         .then(data => this.props.createQuestionSuccess(data))
         .catch(error => this.props.createQuestionError(error))
   };

   createQuestionFormSubmission(event) {
      event.preventDefault();
      const {title, text, prize} = this.state;
      const question = {title, text, prize};
      this.createQuestion(question);
   };

   state = {
     title: '',
     text: '',
     prize: 0
   };

   change(e){
      this.setState({
         [e.target.name]: e.target.value
      })
   };

   render() {
      return (
         <>
            <div className="createQuestionFormContainer">
               <Form onSubmit={this.createQuestionFormSubmission} className="createQuestionForm">
                  <FormGroup className="formGroup">
                     <Label for="title">title</Label>
                     <Input value={this.state.title}
                            onChange={this.change}
                            className="formGroup formGroup_input"
                            type="text" name="title" id="title"
                            placeholder="write a question title"/>
                  </FormGroup>
                  <FormGroup className="formGroup">
                     <Label for="text">text</Label>
                     <Input value={this.state.text}
                            onChange={this.change}
                            className="formGroup formGroup_input"
                            type="textarea"
                            name="text"
                            id="text"
                            rows={5}
                            placeholder="write a description"/>
                  </FormGroup>
                  <FormGroup className="formGroup">
                     <Label for="prize">prize</Label>
                     <Input value={this.state.prize}
                            onChange={this.change}
                            className="formGroup formGroup_input" type="text" name="prize" id="prize"
                            placeholder="write a question title"/>
                  </FormGroup>
                  <Button>Submit</Button>
               </Form>
            </div>
         </>
      )
   }
}

const mapDispatchToProps = dispatch => ({
   createQuestionError: (error) => dispatch(createQuestionError(error)),
   createQuestionSuccess: (json) => dispatch(createQuestionSuccess(json)),
   createQuestionBegin: () => dispatch(createQuestionBegin()),
});
const mapStateToProps = state => ({
   loading: state.createQuestionState.loading,
   error: state.createQuestionState.error
});
export default connect(mapStateToProps, mapDispatchToProps)(CreateQuestion);
