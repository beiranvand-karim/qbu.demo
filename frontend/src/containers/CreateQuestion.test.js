import {shallow} from 'enzyme'
import React from "react"
import CreateQuestion from "./CreateQuestion";
import renderer from 'react-test-renderer'

it ('should render without crashing', () => {
  shallow(<CreateQuestion/>)
});

it('should matches the snapshot', () => {
  const tree = renderer.create(<CreateQuestion />).toJSON();
  expect(tree).toMatchSnapshot()
});
