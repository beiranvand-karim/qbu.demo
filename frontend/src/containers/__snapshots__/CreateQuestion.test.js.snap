// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matches the snapshot 1`] = `
<h1>
  create question page
</h1>
`;
